installer/upgrader/Migrations/V5/LangStringHelper.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Tools\Migrations\V5;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Query\QueryBuilder;
+use OrangeHRM\Core\Traits\ORM\EntityManagerHelperTrait;
+use Symfony\Component\Yaml\Yaml;
+
+class LangStringHelper
+{
+    use EntityManagerHelperTrait;
+
+    /**
+     * @param int $groupId
+     * @return LangString[]
+     */
+    public function getLangStringArray(string $groupName): array
+    {
+        $groupId = $this->getGroupId($groupName);
+        $langArray = [];
+        $filepath = 'installer/upgrader/Migrations/V5/' . $groupName . 'LangString.yaml';
+        $yml = Yaml::parseFile($filepath);
+        $langStrings = array_shift($yml);
+        foreach ($langStrings as $langString) {
+            $langArray[] = new LangString($langString['unitId'], $groupId, $langString['value'], null, null);
+        }
+        return $langArray;
+    }
+
+    /**
+     * @param string $moduleName
+     * @return int
+     * @throws Exception
+     */
+    public function getGroupId(string $moduleName): int
+    {
+        $q = $this->createQueryBuilder();
+        $q->select('module.id')->from('ohrm_i18n_group', 'module')->where('module.name = :group')->setParameter('group', $moduleName);
+        return $q->executeQuery()->fetchOne();
+    }
+
+    /**
+     * @return QueryBuilder
+     */
+    protected function createQueryBuilder(): QueryBuilder
+    {
+        return $this->getEntityManager()->getConnection()->createQueryBuilder();
+    }
+
+    /**
+     * @param int $groupId
+     * @return void
+     * @throws Exception
+     */
+    public function deleteNonCustomLangStrings(int $groupId): void
+    {
+        $groupStrings = $this->getLangStringRecords($groupId);
+        $q = $this->createQueryBuilder();
+        $q->delete('ohrm_i18n_translate')->where('ohrm_i18n_translate.customized != 1')->andWhere($q->expr()->in('ohrm_i18n_translate.lang_string_id', ':groupIds'))->setParameter('groupIds', $groupStrings, Connection::PARAM_INT_ARRAY)->executeQuery();
```

Comment:
@samanthajayasinghe please check this file. Based on Doctrine DBAL, this is how I'm going to propose for upgrader. Need to check Doctrine DBAL limitations